drivers/mysql/mysql_test.go

```diff
@@ -170,29 +144,23 @@ func (suite *MysqlTestSuite) TestCreateSchemaTableIfNotExists() {
 		driver, err := drivers.GetDriver(driverName)
 		suite.Require().NoError(err, "fetching already registered driver should not fail")
 
-		err = driver.CreateSchemaTableIfNotExists()
+		_, err = driver.AppliedMigrations()
 		suite.Assert().Error(err, "should error when database connection is missing")
 		suite.Assert().EqualError(err, "driver: mysql, message: database connection is missing, originalError: driver has no connection established ")
 	})
 
 	suite.T().Run("when x-migrations-table is missing, it creates a migrations table if not exists based on the default configuration", func(t *testing.T) {
-		driver, err := drivers.GetDriver(driverName)
-		suite.Require().NoError(err, "fetching already registered driver should not fail")
-
-		_, err = driver.Open(testConnURL)
-		suite.Assert().NoError(err, "should not error when connecting to database from url")
-		defer func() {
-			err = driver.Close()
-			suite.Require().NoError(err, "should not error when closing the database connection")
-		}()
+		connectedDriver, teardown := suite.InitializeDriver(testConnURL)
+		defer teardown()
 
-		_, err = suite.testDB.Exec(fmt.Sprintf("DROP TABLE IF EXISTS %s", defaultConfig.MigrationsTable))
+		_, err := suite.testDB.Exec(fmt.Sprintf("DROP TABLE IF EXISTS %s", defaultConfig.MigrationsTable))
 		suite.Require().NoError(err, "should not error while dropping pre-existing migrations table")
 
 		migrationTableExists := fmt.Sprintf(`SELECT COUNT(*) FROM information_schema.tables
 								WHERE  table_schema = '%s'
 								AND    table_name = '%s';`, databaseName, defaultConfig.MigrationsTable)
-		err = driver.CreateSchemaTableIfNotExists()
+
+		_, err = connectedDriver.AppliedMigrations()
```

Comment:
I see the problem here. We are calling the exported method just for side-effect. Fine for now, but the test suite can be improved a bit. We are inside the package, so it feels a bit weird to go via the interface here. To test internal methods, we should just go via the concrete struct.

Anyways, out of scope for this PR.

Reply:
I actually thought about this and decided to stick on public API for testing but 0/5 on either.